In [1]:
import numpy as np
import pandas as pd

In [27]:
from sklearn.model_selection import cross_val_score
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import StandardScaler

## Read in data

In [4]:
df_red = pd.read_csv('./dataset/winequality-red.csv', sep=';', index_col=None)
df_white = pd.read_csv('./dataset/winequality-white.csv', sep=';', index_col=None)

In [5]:
X_red = df_red.iloc[:, :-1].values
y_red = df_red.iloc[:, -1].values
X_white = df_white.iloc[:, :-1].values
y_white = df_white.iloc[:, -1].values

In [22]:
scaler = StandardScaler()

## Choose random forest classifier

In [24]:
clf = RandomForestClassifier(max_depth=5, random_state=0)
print("For red wine, the mean accuracy is: ", cross_val_score(clf, scaler.fit_transform(X_red), y_red, cv=5).mean())

For red wine, the mean accuracy is:  0.5928859717868338


In [26]:
clf = RandomForestClassifier(max_depth=5, random_state=0)
print("For white wine, the mean accuracy is: ", cross_val_score(clf, scaler.fit_transform(X_white), y_white, cv=5).mean())

For white wine, the mean accuracy is:  0.5216474536699256


The random forest classifier has 100 trees with each max depth of tree as 5.
The performance from a 5-fold cross-validation is 0.5929 for the red wine and 0.5216 for the white wine, which are sub-optimal.
To improve the performance, a more careful analysis of the dataset is required to understand the distribution of the sample in each feature space.